### DISCUSSION TITLE
How to use 'setting'?

### DISCUSSION THREAD

**Discussion by fermevc** [spatie/laravel-settings#70] [Q&A] [answered]
My initial thought was to use this package to persist various App settings separated in appropriate groups (Email Account Settings, Database Connection Settings, ...). Everything was going in right direction until I came to the point when I wanted to use the settings stored in database. By following the examples, I was able to integrate "show" and "update" logic for three separate setting groups (simple UI using Vue JS). All variables for each group are properly read from DB, and I can successfully edit and update any of them.

How should one use these stored settings in place of variables found in ENV or CONFIG files?

Most simple example that I can think of is:
Set APP_NAME to the value stored in database using this package. For example, in the UI interface, when administrator updates App name, set that value for every request.

I've tried to use
`app(AppSettings::class)->name`
but that only works if called from controller.
My current PHP/Laravel level prevents me to try and figure this on my own. 

Thanks in advance!

**Comment by rubenvanassche** [comment, marked as answer]
Hi @fermevc,

I haven't tried such approach with overwriting config and .env values, though it should be possible. You can create a `SettingsProvider` (https://laravel.com/docs/8.x/providers) which would overwrite the config values when it is booted. Not totally sure if that would completely work. But I think it's a good starting point for solving this issue.

**Reply by fermevc** [reply]
Thank you @rubenvanassche .
I've managed to, temporarily, solve my problem by using/resolving "setting" classes in custom middleware, where I'm doing some checks to see what've been set (in setting classes, .env or config)... If something is set in "setting" UI, I override appropriate setting in config, and it will be in use on the next request... Not the best solution, a lot of manual checks, but I'm happy at the moment. Thanks for all the help and all the good work!
